tests/kafkatest/tests/replication_test.py

```diff
@@ -75,28 +116,35 @@ def run_with_failure(self, failure):
         indicator that nothing is left to consume.
 
         """
-        self.producer = VerifiableProducer(self.test_context, self.num_producers, self.kafka, self.topic, throughput=self.producer_throughput)
-        self.consumer = ConsoleConsumer(self.test_context, self.num_consumers, self.kafka, self.topic, consumer_timeout_ms=3000)
+        self.num_bounce = num_bounce
+        for producer in self.producers:
+            producer.producer_props = producer_props.copy()
 
         # Produce in a background thread while driving broker failures
-        self.producer.start()
-        if not wait_until(lambda: self.producer.num_acked > 5, timeout_sec=5):
-            raise RuntimeError("Producer failed to start in a reasonable amount of time.")
-        failure()
-        self.producer.stop()
+        self.logger.debug("Producing messages...")
+        self.start_producers()
+
+        self.logger.debug("Driving failures...")
+        self.drive_failures(failure, node_type)
+        time.sleep(5)  # Keep on producing for a few more seconds
```

Comment:
Why is this necessary? Do we not wait long enough during `drive_failures` to make sure we're back in a good state?

Would be nice to keep the arbitrary `sleep()` calls to a minimum :)